### PR TITLE
LOOP-1155 Implement reservoir view in SwiftUI

### PR DESCRIPTION
**Pull request by ps2**
https://tidepool.atlassian.net/browse/LOOP-1155

* Implements reservoir view for omnipod settings page in SwiftUI
* Removes DashSettingsViewModelProtocol, now that we have a working MockPodCommManager we can use for previews, testing, etc.
* Update pumpmanager title to "Omnipod 5"
* Mock Pod Settings now accessed by tapping on pod image, rather than having separate menu item. This allows QA/marketing to generate screenshots that are more true to actual use.

### CODE REVIEW

File: DashKit/ReservoirLevel.swift [hunk 1]
```diff
@@ -17,12 +17,13 @@ public enum ReservoirLevel: RawRepresentable, Equatable {
     case valid(Double)
     case aboveThreshold
 
-    public func asPercentage() -> Double? {
+    public var percentage: Double {
         switch self {
         case .aboveThreshold:
-            return nil
+            return 1
```
Comment (nhamming) [inline DashKit/ReservoirLevel.swift:23]: +1 At least I was confused by this at one point in time.

File: DashKit/ReservoirLevel.swift [hunk 2]
```diff
@@ -17,12 +17,13 @@ public enum ReservoirLevel: RawRepresentable, Equatable {
...
         case .valid(let value):
-            return min(1, max(0, value / Pod.reservoirCapacity))
+            // Set 50U as the halfway mark, even though pods can hold 200U.
+            return min(1, max(0, value / 100))
```
Comment (nhamming) [inline DashKit/ReservoirLevel.swift:26]: Cool. I'll delete my PR related to this.

File: DashKitUI/Views/DashSettingsView.swift [hunk 3]
```diff
@@ -281,7 +323,7 @@ struct DashSettingsView<Model>: View where Model: DashSettingsViewModelProtocol
         .alert(isPresented: $viewModel.alertIsPresented, content: { alert(for: viewModel.activeAlert!) })
         .insetGroupedListStyle()
         .navigationBarItems(trailing: doneButton)
-        .navigationBarTitle("Omnipod", displayMode: .automatic)
+        .navigationBarTitle("Omnipod 5", displayMode: .automatic)
```
Comment (nhamming) [inline DashKitUI/Views/DashSettingsView.swift]: Should this come from the viewModel?

File: DashKitUI/Views/DashSettingsView.swift [hunk 4]
```diff
@@ -333,6 +375,33 @@ struct DashSettingsView<Model>: View where Model: DashSettingsViewModelProtocol
...
+            return insulinTintColor
+        case .valid(let value):
+            if value > 10 {
+                return insulinTintColor
+            } else {
+                return guidanceColors.warning
+            }
+        }
+    }
 }
```
Comment (nhamming) [inline DashKitUI/Views/DashSettingsView.swift]: [nit] Should this logic live in the view model that just exposes a `String` or `Color`?

**Review by nhamming — APPROVED** [review]
No blockers.

File: DashKitUI/Views/DashSettingsView.swift (hunk 4 above)
Comment (ps2) [reply DashKitUI/Views/DashSettingsView.swift]: I moved reservoirText, but reservoirColor depends on Environment, so I kept it here.

**Pull request merged**